toonz/sources/toonz/filmstripcommand.cpp

```diff
@@ -51,6 +53,53 @@ TFrameId operator+(const TFrameId &fid, int d) {
 
 //-----------------------------------------------------------------------------
 
+void updateXSheet(TXshSimpleLevel *sl, std::vector<TFrameId> oldFids,
+                  std::vector<TFrameId> newFids, TXsheet *xsh) {
+  if (!Preferences::instance()->isSyncLevelRenumberWithXsheetEnabled()) return;
```

Comment:
For now `updateXsheet()` checks the current preference value of `Sync Level Strip Drawing Number Changes with the Xsheet` before execution.

Which means that, if you toggle the preference value just after editing the level, undo will not revert the scene properly. I think this value should be remembered by each undo instance.